src/sentry/replays/endpoints/project_replay_summarize_breadcrumbs.py

```diff
@@ -137,15 +162,19 @@ def generate_error_log_message(error: ErrorEvent) -> str:
 
 
 def get_request_data(
-    iterator: Iterator[tuple[int, memoryview]], error_events: list[ErrorEvent]
+    iterator: Iterator[tuple[int, memoryview]],
+    error_events: list[GroupEvent],
+    project_id: int | None = None,
```

Comment:
```suggestion
    project_id: int,
```
Do we expect to call this with no project info in the future? Same with gen_request_data and as_log

Reply:
it's to avoid having to mock the project in the unit tests. the project id in these methods is actually only used in the case of fetching feedback details which we know it should exist

Reply:
Why does specifying the project-id require a query?  I don't see an if statement guarding any queries.

Reply:
@aliu39 @cmanallen i updated the PR to make `project_id` required!

Reply:
do you mean remove feedback from the `as_log_message` function? i can do that in `gen_request_data` request, where the call to `as_log_message` happens. but i feel this will result in more complicated/convoluted code. we'll still require the passing of `project_id` to the `get_request_data` function (which calls `gen_request_data`)